cli/connector/eval_plain_text.go

```diff
@@ -0,0 +1,389 @@
+	evalFunc, err := templates.GetTemplatedStr(&callFuncTmpl, map[string]string{
+		"FunctionName": funcName,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to instantiate call function template: %s", err)
+	}
+
+	return evalPlainTextConnYAML(conn, evalFunc, args, opts)
+}
+
+// evalPlainTextConnYAML calls function on Tarantool instance
+// Function should return `interface{}`, `string` (res, err)
+// to be correctly processed.
+// Processes only YAML output.
+func evalPlainTextConnYAML(conn net.Conn, funcBody string, args []interface{}, opts EvalPlainTextOpts) ([]interface{}, error) {
+	if err := formatAndSendEvalFunc(conn, funcBody, args, evalFuncYAMLTmpl); err != nil {
+		return nil, err
+	}
+
+	// recv from socket
+	resBytes, err := readFromPlainTextConnYAML(conn, opts)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to check returned data: %s", err)
+	}
+
+	data, err := processEvalTarantoolResYAML(resBytes, opts.ResData)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+// func evalPlainTextConnLua(conn net.Conn, funcBody string, args []interface{}, opts EvalPlainTextOpts) (interface{}, error) {
+// 	if err := formatAndSendEvalFunc(conn, funcBody, args, evalFuncYAMLTmpl); err != nil {
+// 		return nil, err
+// 	}
+
+// 	// recv from socket
+// 	resBytes, err := readFromPlainTextConnLua(conn, opts)
+// 	if err != nil {
+// 		return nil, fmt.Errorf("Failed to check returned data: %s", err)
+// 	}
+
+// 	res, err := processEvalTarantoolResLua(resBytes)
+// 	if err != nil {
+// 		return nil, fmt.Errorf("Failed to parse data returned from instance: %s", err)
+// 	}
+
+// 	if !res.Success {
+// 		return nil, fmt.Errorf(res.ErrStr)
+// 	}
+
+// 	return res.Data, nil
+// }
+
+func formatAndSendEvalFunc(conn net.Conn, funcBody string, args []interface{}, evalFuncTmpl string) error {
+	if args == nil {
+		args = []interface{}{}
+	}
+
+	argsEncoded, err := msgpack.Marshal(args)
+	if err != nil {
+		return fmt.Errorf("Failed to encode args: %s", err)
+	}
+
+	evalFunc, err := templates.GetTemplatedStr(&evalFuncTmpl, map[string]string{
+		"FunctionBody": funcBody,
+		"ArgsEncoded":  fmt.Sprintf("%x", argsEncoded),
+	})
+
+	if err != nil {
+		return fmt.Errorf("Failed to instantiate eval function template: %s", err)
+	}
+
+	evalFuncFormatted := strings.Join(
+		strings.Split(strings.TrimSpace(evalFunc), "\n"), " ",
+	)
+	evalFuncFormatted = strings.Join(strings.Fields(evalFuncFormatted), " ") + "\n"
+
+	// write to socket
+	if err := writeToPlainTextConn(conn, evalFuncFormatted); err != nil {
+		return fmt.Errorf("Failed to send eval function to socket: %s", err)
+	}
+
+	return nil
+}
+
+func writeToPlainTextConn(conn net.Conn, data string) error {
+	writer := bufio.NewWriter(conn)
+	if _, err := writer.WriteString(data); err != nil {
+		return fmt.Errorf("Failed to send to socket: %s", err)
+	}
+
+	if err := writer.Flush(); err != nil {
+		return fmt.Errorf("Failed to flush: %s", err)
```

Comment:
It would be great some details. "Failed to flush" says nothing for user.

Reply:
For the end user it would be smth like `Failed to send eval function to socket: Failed to flush`.